### ISSUE TITLE
Plugin shutdown on Windows can hang

### ISSUE THREAD

**Issue by joeduffy** [pulumi/pulumi#962]
We encountered an issue where plugin shutdown can hang on Windows.  (Specifically, this was the NodeJS language plugin.)  The problem seems to be that failures at certain points may not cause the stdout channel to get closed.  I debugged far enough along to witness the following:

1. `stdoutDone` channel never gets closed, because the `ReadLine` never completes
2. killing processes fails with `Access Denied` errors, because the processes are gone
3. because of (1), the wait on the `stdoutDone` channel hangs indefinitely

I triggered this by accident because we had a `${VERSION}` in the NodeJS SDK's `version.js` file, rather than a valid semantic version.  This meant that during plugin spin-up we had a version that was invalid, triggering an error "earlier" than expected.  For example:

1. go to a simple example, like `examples/dynamic-provider/simple/`
2. edit `./node_modules/@pulumi/pulumi/version.js`, and change the version string to `${INVALID}`
3. run pulumi preview

Interestingly, this fails and terminates the program correctly on Mac:

```
error: An error occurred while advancing the preview: Invalid character(s) found in major number "${INVALID}"
```

But on Windows, this hangs each and every time.

I don't believe this is a new issue and suspect that it has been the case for a while that if you fail at just the right time, we will end up hanging during premature plugin shutdown on Windows.

**Comment by stack72** [comment]
I am going to close this out - it's been 3 years since it was opened and hasn't become a widespread thing - we will monitor and see if this is something users still hit
